src/taskpane/taskpane.ts

```diff
@@ -29,6 +30,22 @@ Office.onReady((info) => {
     document.querySelectorAll(".logo-button").forEach((button) => {
       (button as HTMLElement).onclick = () => insertImageByBase64(button.getAttribute("data-value"));
     });
+
+    document.getElementById("icons").onclick = async () => {
+      document.querySelectorAll(".icon-results img").forEach((img) => img.remove());
+
+      try {
+        const searchTerm = (<HTMLInputElement>document.getElementById("icon-search-input")).value;
+        const result = await fetchIcons(searchTerm);
+        result.forEach((obj) => {
+          getImageElementWithSource(obj.id, obj.url);
+        });
+      } catch (e) {
+        throw new Error("Error retrieving icon urls: " + e);
+      }
+    };
+
+    document.getElementById("icon-urls").addEventListener("click", (event) => insertBase64ImageOn(event), false);
```

Comment:
Mein Plan war in unserer "Main" Methode nur Einzeiler zu lassen, ich denke die Stelle soll so sauber wie möglich sein. Denkst du wir können/sollen diesen block in eine separate Methode auslagern?

Reply:
wir können das natürlich separat bei einer Refactoring machen

Reply:
Klar, du hast recht. Wurde jetzt in einzelne Methoden extrahiert, aber da ist sicher noch Platz nach oben im Refactoring.